src/include/sof/audio/volume.h

```diff
@@ -277,4 +280,110 @@ static inline void peak_vol_update(struct vol_data *cd)
 void sys_comp_module_volume_interface_init(void);
 #endif
 
+vol_zc_func vol_get_zc_function(struct comp_dev *dev,
+				struct comp_buffer __sparse_cache *sinkb);
+
+#if CONFIG_COMP_VOLUME_LINEAR_RAMP
+/**
+ * \brief Calculate linear ramp function
+ * \param[in,out] dev Component data: ramp start gain, actual gain
+ * \param[in] ramp_time Time spent since ramp start as milliseconds Q29.3
+ * \param[in] channel Current channel to update
+ */
+static inline int32_t volume_linear_ramp(struct vol_data *cd, int32_t ramp_time, int channel)
+{
+	return cd->rvolume[channel] + ramp_time * cd->ramp_coef[channel];
+}
+#endif
+
+#if CONFIG_COMP_VOLUME_WINDOWS_FADE
+/**
+ * \brief Calculate windows fade ramp function
+ * \param[in,out] dev Component data: target gain, ramp start gain, ramp duration
+ * \param[in] ramp_time Time spent since ramp start as milliseconds Q29.3
+ * \param[in] channel Current channel to update
+ */
+
+static inline int32_t volume_windows_fade_ramp(struct vol_data *cd, int32_t ramp_time, int channel)
+{
+	int32_t time_ratio; /* Q2.30 */
+	int32_t pow_value; /* Q2.30 */
+	int32_t volume_delta = cd->tvolume[channel] - cd->rvolume[channel]; /* Q16.16 */
+
+	if (!cd->initial_ramp)
+		return cd->tvolume[channel];
+
+	time_ratio = (((int64_t)ramp_time) << 30) / (cd->initial_ramp << 3);
+	pow_value = volume_pow_175(time_ratio);
+	return cd->rvolume[channel] + Q_MULTSR_32X32((int64_t)volume_delta, pow_value, 16, 30, 16);
+}
+#endif
+
+void reset_state(struct vol_data *cd);
```

Comment:
`volume_reset_state()`

Reply:
I did want to change like add volume_ prefix, but later find there is no extra errors, so did not make it, let me change.